File: Ten Days Of Javascript/d2_string_basics.js
/* Solutions for

https://www.hackerrank.com/challenges/js10-switch
https://www.hackerrank.com/challenges/js10-loops

*/

/* eslint-disable no-console */
function firstOfSet(s) {
  const letter = s.charAt(0);
  let output;
  switch (letter) {
    // Using ? so that if no match is found, undefined is returned (so false)
    case letter.match('[aeiou]')?.[0]:
      output = 'A';
      break;
    case letter.match('[bcdfg]')?.input:
      output = 'B';
      break;
    case letter.match('[hjklm]')?.input:
      output = 'C';
      break;
    case letter.match('[npqrstvwxyz]')?.input:
      output = 'D';
      break;
    default:
      output = '-1';
  }

  console.log(output);
  return output;
}

firstOfSet('manuel');

function vowelsAndConsonants(s) {
  const vowels = [];
  const cnsonts = [];
  s.split('').forEach((x) => {
    if (x === 'a' || x === 'e' || x === 'i' || x === 'o' || x === 'u') {
      vowels.push(x);
    } else {
      cnsonts.push(x);
    }
  });

  const all = vowels.concat(cnsonts);
  all.forEach((x) => console.log(x));
}

vowelsAndConsonants('manuel');

process.exit();

File: Ten Days of Python/d2_string_basics.py
"""
# https://www.hackerrank.com/challenges/js10-switch
# https://www.hackerrank.com/challenges/js10-loops

"""


def first_of_set(s: str):
    letter = s[0]
    result = None

    if letter in "aeiou":
        result = "A"
    elif letter in "bcdfg":
        result = "B"
    elif letter in "hjklm":
        result = "C"
    elif letter in "npqrstvwxyz":
        result = "D"

    return result


def vowels_and_consonants(s: str):

    vowels = []
    consonants = []

    for letter in s:
        if letter in "aeiou":
            vowels.append(letter)
        else:
            consonants.append(letter)

    for el in vowels + consonants:
        print(el)

    return


vowels_and_consonants("manuel")
